lib/transfer.c

```diff
@@ -1232,6 +1226,12 @@ CURLcode Curl_readwrite(struct connectdata *conn,
     return CURLE_SEND_ERROR;
   }
 
+  if(data->state.drain) {
+    data->state.drain--;
+    select_res |= CURL_CSELECT_IN;
+    DEBUGF(infof(data, "Curl_readwrite: forcibly told to drain data\n"));
```

Comment:
First, I just filed PR #5045 that again removes that drain counter decrement.

Then, by moving this check you now let an error from select() override draining data, which I'm not entirely sure is a good idea.